pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -12176,6 +12224,349 @@ spec:
+    pipelineTaskName: matrix-with-onerror
+  provenance:
+    featureFlags:
+      RunningInEnvWithInjectedSidecars: true
+      EnableTektonOCIBundles: true
+      EnableAPIFields: "alpha"
+      EnforceNonfalsifiability: "none"
+      AwaitSidecarReadiness: true
+      VerificationNoMatchPolicy: "ignore"
+      EnableProvenanceInStatus: true
+      ResultExtractionMethod: "termination-message"
+      MaxResultSize: 4096
+      Coschedule: "workspaces"
+`),
+	}, {
+		name:     "p-finally",
+		memberOf: "finally",
+		p: parse.MustParseV1Pipeline(t, fmt.Sprintf(`
+metadata:
+  name: %s
+  namespace: foo
+spec:
+  tasks:
+    - name: unmatrixed-pt
+      params:
+        - name: param-1
+          value: "100"
+      taskRef:
+        name: mytask
+  finally:
+    - name: matrix-with-onerror
+      taskRef:
+        name: mytask
+      matrix:
+        params:
+          - name: param-1
+            value:
+              - "0"
+              - "1"
+              - "2"
+              - "3"
+`, "p-finally")),
+		tr: mustParseTaskRunWithObjectMeta(t,
+			taskRunObjectMeta("pr-unmatrixed-pt", "foo",
+				"pr", "p-finally", "unmatrixed-pt", false),
+			`
+spec:
+  params:
+  - name: param-1
+    value: "100"
+  serviceAccountName: test-sa
+  taskRef:
+    name: mytask
+    kind: Task
+status:
+ conditions:
+  - type: Succeeded
+    status: "True"
+    reason: Succeeded
+    message: All Tasks have completed executing
+`),
+		expectedPipelineRun: parse.MustParseV1PipelineRun(t, `
+metadata:
+  name: pr
+  namespace: foo
+  annotations: {}
+  labels:
+    tekton.dev/pipeline: p-finally
+spec:
+  taskRunTemplate:
+    serviceAccountName: test-sa
+  pipelineRef:
+    name: p-finally
+status:
+  pipelineSpec:
+    tasks:
+    - name: unmatrixed-pt
+      params:
+        - name: param-1
+          value: "100"
+      taskRef:
+        name: mytask
+        kind: Task
+    finally:
+    - name: matrix-with-onerror
+      taskRef:
+        name: mytask
+        kind: Task
+      matrix:
+        params:
+          - name: param-1
+            value:
+              - "0"
+              - "1"
+              - "2"
+              - "3"
+  conditions:
+  - type: Succeeded
+    status: "Unknown"
+    reason: "Running"
+    message: "Tasks Completed: 1 (Failed: 0, Cancelled 0), Incomplete: 1, Skipped: 0"
+  childReferences:
+  - apiVersion: tekton.dev/v1
+    kind: TaskRun
+    name: pr-unmatrixed-pt
+    pipelineTaskName: unmatrixed-pt
+  - apiVersion: tekton.dev/v1
+    kind: TaskRun
+    name: pr-matrix-with-onerror-0
+    pipelineTaskName: matrix-with-onerror
+  - apiVersion: tekton.dev/v1
+    kind: TaskRun
+    name: pr-matrix-with-onerror-1
+    pipelineTaskName: matrix-with-onerror
+  - apiVersion: tekton.dev/v1
+    kind: TaskRun
+    name: pr-matrix-with-onerror-2
+    pipelineTaskName: matrix-with-onerror
+  - apiVersion: tekton.dev/v1
+    kind: TaskRun
+    name: pr-matrix-with-onerror-3
+    pipelineTaskName: matrix-with-onerror
+  provenance:
```

Comment:
non-blocking: I wonder if it makes sense to ignore the provenance fields for these tests

Reply:
sure @dibyom I will open a separate PR to clean up provenance fields as its part of many other tests.